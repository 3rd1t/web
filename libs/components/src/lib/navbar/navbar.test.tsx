import React from "react"
import { cleanup, fireEvent, render } from "@testing-library/react"
import renderer from "react-test-renderer"
import Navbar from "./navbar"


afterEach(cleanup)

describe(" Navbar", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Navbar
          logo={<span>I am a logo</span>}
          links={[
            {
              label: "Product",
              href: "#product",
              internal: true,
            },
            {
              label: "Features",
              href: "#features",
              internal: true,
            },
            {
              label: "Team",
              href: "#team",
              internal: true,
            },
            {
              label: "Documentation",
              href: "#",
              internal: false,
            },
          ]}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe("on mobile", () => {
    describe("when the user clicks on the hamburger icon", () => {
      it("opens the navbar menu", async () => {
        const { queryByTestId } = render(
          <Navbar
            logo={<span>Logo</span>}
            links={[
              {
                label: "Product",
                href: "#product",
                internal: true,
              },
              {
                label: "Features",
                href: "#features",
                internal: true,
              },
              {
                label: "Team",
                href: "#team",
                internal: true,
              },
              {
                label: "Documentation",
                href: "#",
                internal: false,
              },
            ]}
          />,
        )

        expect(queryByTestId("mobileNavbar")).toBeNull()

        fireEvent.click(queryByTestId("toggle"))
        expect(queryByTestId("mobileNavbar")).toBeTruthy()
      })
    })
  })
})
