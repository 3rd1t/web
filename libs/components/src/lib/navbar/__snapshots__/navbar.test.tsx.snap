// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Navbar renders correctly 1`] = `
<nav
  className="container flex flex-wrap items-center justify-between p-4 mx-auto md:flex-no-wrap"
  role="navigation"
>
  <a
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <span>
      I am a logo
    </span>
  </a>
  <div
    className="md:hidden"
  >
    <button
      aria-label="home"
      className="p-2 text-gray-400 hover:text-gray-900 focus:outline-none"
      data-testid="toggle"
      onClick={[Function]}
    >
      <svg
        className="w-6 h-6"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
      >
        <path
          d="M4 6h16M4 12h16M4 18h16"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
  <section
    className="hidden w-auto md:block"
  >
    <ul
      className="flex items-center space-x-10"
      data-testid="desktopNavbar"
    >
      <li>
        <a
          className="text-lg font-medium text-gray-700 transition duration-150 ease-in-out hover:text-black"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Product
        </a>
      </li>
      <li>
        <a
          className="text-lg font-medium text-gray-700 transition duration-150 ease-in-out hover:text-black"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Features
        </a>
      </li>
      <li>
        <a
          className="text-lg font-medium text-gray-700 transition duration-150 ease-in-out hover:text-black"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          About us
        </a>
      </li>
      <li>
        <a
          className="text-lg font-medium text-gray-700 transition duration-150 ease-in-out hover:text-black"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Documentation
        </a>
      </li>
    </ul>
  </section>
</nav>
`;
